src/columns.jl

```diff
@@ -314,7 +314,14 @@ end
 
 function permute!(c::Columns, p::AbstractVector)
     for v in c.columns
-        copy!(v, v[p])
+        if isa(v, StringArray{String})
+            ws = convert(StringArray{WeakRefString{UInt8}}, v)
+            permute!(ws, p)
+        elseif isa(v, PooledArrays.PooledArray)
+            permute!(v, p)
+        else
+            copy!(v, v[p])
```

Comment:
I don't know why this was the implementation....

Shouldn't `permute!(v, p)` work in the general case?

Reply:
I think because:
```
julia> using BenchmarkTools

julia> a = rand(Int, 10^6);

julia> p = shuffle!([1:10^6;]);

julia> @btime permute!(a, p);
  65.502 ms (2 allocations: 7.63 MiB)

julia> @btime copy!(a, a[p]);
  9.788 ms (3 allocations: 7.63 MiB)
```

Reply:
makes sense it's probably because the cache keeps getting invalidated on every write if you don't use a separate buffer?